rasa/train.py

```diff
@@ -196,17 +197,29 @@ def train(
     force_training: bool = False,
     retrain_core: bool = True,
     retrain_nlu: bool = True,
+    replace_templates: bool = True,
     fixed_model_name: Optional[Text] = None,
     persist_nlu_training_data: bool = False,
     kwargs: Optional[Dict] = None,
 ):
 
-    if force_training or retrain_core:
+    if retrain_core or replace_templates or force_training:
+        replace_templates_only = (
+            replace_templates and not retrain_core and not force_training
+        )
+        if replace_templates_only:
```

Comment:
I think instead of running the agent training we should only persist the templates here, avoiding to push the flag to agent train